ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/PrivateTransactionProcessor.java

```diff
@@ -83,24 +82,22 @@ public PrivateTransactionProcessor(
     this.privateTransactionValidator = privateTransactionValidator;
   }
 
-  @SuppressWarnings("unused")
   public TransactionProcessingResult processTransaction(
-      final Blockchain blockchain,
       final WorldUpdater publicWorldState,
       final WorldUpdater privateWorldState,
-      final ProcessableBlockHeader blockHeader,
+      final BlockHeader blockHeader,
       final Hash pmtHash,
       final PrivateTransaction transaction,
       final Address miningBeneficiary,
       final OperationTracer operationTracer,
-      final BlockHashLookup blockHashLookup,
+      final Function<Long, Hash> blockHashLookup,
       final Bytes privacyGroupId) {
     try {
       LOG.trace("Starting private execution of {}", transaction);
 
       final Address senderAddress = transaction.getSender();
-      final EvmAccount maybePrivateSender = privateWorldState.getAccount(senderAddress);
-      final MutableAccount sender =
+      final var maybePrivateSender = privateWorldState.getAccount(senderAddress);
```

Comment:
Is type inference necessary here? I'm not seeing what it is buying us here, personally I think having the type declarations here makes the following ternary operation more readable.

Reply:
Ease of writing.  I'll adjust it.